### PR TITLE
Add get_level_values for Index and MultiIndex

### PR DESCRIPTION
**Pull request by charlesdong1991**
This PR is blocked by #973 where `levels` property is implemented, and by #971 where seems @itholic is into fixing it soon ^^

^ above is done!!

```python
mi = pd.MultiIndex.from_arrays((list('abc'), list('def')))
mi.names = ['level_1', 'level_2']
kdf = ks.DataFrame({'a': [1, 2, 3]}, index=mi)

# get values by level number
kdf.index.get_level_values(0)

# get values by level number
kdf.index.get_level_values('level_1')
```

### CODE REVIEW

**Comment by itholic** [comment]
Ah!! then i think it should be better you take #971 to take care of entire implementation related with this. What r u think?? :smiley:

**Comment by charlesdong1991** [comment]
no worries, they are separate issues IMO. Even if i work on this, I will make a separate PR anyway. 

Do you want to work on it? @itholic I am asking because I will take long-haul flight soon, not sure if I  will have time and energy for this in next two days, so if you are into it, pls go ahead! @itholic

**Comment by itholic** [comment]
@charlesdong1991 Ah, yes i get it. then im going to take that one. Hope you having a great flight :clap:

**Comment by charlesdong1991** [comment]
thanks, mate! @itholic 😄

**Comment by HyukjinKwon** [comment]
This one needs @ueshin's review .. he knows index stuff much better then I do.

**Comment by itholic** [comment]
FYI, resolved #971 , PR underway #948 😄

**Comment by charlesdong1991** [comment]
haha! very productive and nice work!! @itholic 👍

**Comment by charlesdong1991** [comment]
i think this PR is ready for review now @ueshin

File: databricks/koalas/indexes.py [hunk 1]
```diff
@@ -426,6 +426,25 @@ def _validate_index_level(self, level):
...
+        Index(['a', 'b', 'c'], dtype='object', name='ks')
+
+        >>> kdf = ks.DataFrame({'a': [1, 2, 3]}, index=pd.Index(['a', 'b', 'c'], name='ks'))
+        >>> kdf.index.get_level_values('ks')
+        Index(['a', 'b', 'c'], dtype='object', name='ks')
+        """
+        self._validate_index_level(level)
+        return self
+
+    get_level_values = _get_level_values
```
Comment (ueshin) [inline databricks/koalas/indexes.py]: Why is this separated? Can't we just `def get_level_values`?

File: databricks/koalas/indexes.py [hunk 2]
```diff
@@ -426,6 +426,25 @@ def _validate_index_level(self, level):
...
             )
 
+    def _get_level_values(self, level):
+        """
+        Return Index if a valid level is given.
+
+        Examples:
+        --------
+        >>> kdf = ks.DataFrame({'a': [1, 2, 3]}, index=pd.Index(['a', 'b', 'c'], name='ks'))
+        >>> kdf.index.get_level_values(0)
```
Comment (ueshin) [inline databricks/koalas/indexes.py]: Now that we can construct `Index`. How about using it?

```py
index = ks.Index(['a', 'b', 'c'], name='ks')
```

File: databricks/koalas/indexes.py [hunk 3]
```diff
@@ -690,6 +709,96 @@ def levels(self) -> list:
...
+
+        Parameters
+        ----------
+        level : int level
+
+        Returns
+        -------
+        values : Index
+        """
+        lev = self.levels[level]
```
Comment (ueshin) [inline databricks/koalas/indexes.py]: We should avoid using `levels` property since it runs a Spark job.

Comment (ueshin) [reply databricks/koalas/indexes.py]: Also the returned value will be different.

```py
>>> pd.MultiIndex.from_tuples([('x', 'a'), ('x', 'b'), ('y', 'a')]).get_level_values(0)
Index(['x', 'x', 'y'], dtype='object')
```

**Comment by HyukjinKwon** [comment]
@itholic can you take this over? We can credit together.

**Comment by itholic** [comment]
Sure, I'll create a PR to finish this work. :)

File: databricks/koalas/indexes.py (hunk 3 above)
Comment (itholic) [reply databricks/koalas/indexes.py:760]: The comments here are resolved at https://github.com/databricks/koalas/pull/1517/commits/dd8683c1fdf6ab9a2f86c822cf6a5fad3c581c04.

**Comment by itholic** [comment]
Hi, @ueshin . 

The comments here are resolved at https://github.com/databricks/koalas/pull/1517/commits/dd8683c1fdf6ab9a2f86c822cf6a5fad3c581c04 in #1517

**Comment by HyukjinKwon** [comment]
Let's track it in #1517

**Pull request closed without merging**